orchestra/analytics/latency.py

```diff
@@ -128,3 +132,55 @@ def project_time_row_generator(project_information,
             [iteration.end_datetime for iteration in iterations])
         for start_datetime, iteration in zip(iteration_times[:-1], iterations):
             yield iteration.to_dict(start_datetime)
+
+
+def work_time_aggregate(df, dimensions):
+    """
+    Aggregates work time grouped by `dimensions`.
+
+    Args:
+        df (pandas.DataFrame):
+            A DataFrame with a schema described in `work_time_df`.
+        dimensions (list):
+            A list of column names to group on.
+
+    Returns:
+        df (pandas.DataFrame):
+            A DataFrame with aggregate information.
+    """
+    return df.groupby(dimensions, as_index=False).agg({'work_time': np.sum})
```

Comment:
To the first two questions: absolutely.  work_time_aggregate's implementation is simple, but calendar_time_aggregate's implementation is not.  Providing the functions encapsulates the challenging logic so that people who want to work with the data have a better abstraction.

To the third question: Pandas `groupby` is more like Pig's grouping operator than SQL's.  It creates indexed/hierarchically clustered keys that map on to the remaining ungrouped columns. `as_index=False` flattens things out so the result is a lot more like a relation/SQL group by.  Is this worth a comment?  I'd have to add it to both places where I group, but it feels a bit idiosyncratic to Pandas.